TalismanSociety/signet-web
# [Feat] - Display referendum Titles, add react-query

# Description

## ⛑️ **What was done:**

- Fetch referendum data from Subsquare API and display referendum titles on Proposal Dropdown and Pending Votes components
- Titles with default to previously existing behavior for unsupported chains or proposals without title

## Technical:

- Installed react-query and react-query devtools for fetching and caching Rest (or GraphQL) apis. [React-query docs](https://tanstack.com/query/latest/docs/framework/react/overview)
- Created reusable custom hook to fetch & referendum data by ID
Note: The API does not support a request with an array of IDs, N requests are required to get the data for each referendum. API calls to unsupported chains are **not** executed, though they how appear on client side cache as `chainId, { isChainSupported: false }, id`

## Type of change

Please delete options that are not relevant.


### 🧪 **How to test:**
1. Visit `/voting`
2. If the chain is supported by Subsquare, the proposals dropdown will show the titles of the proposals
3. Cast a vote
4. If the chain is supported by Subsquare, Pending Votes list will show the correct title
5. Check react-query devtools if you are interested in understanding how the client-side cache works with the query-keys


### 🖼️ **Screenshots:**


https://github.com/TalismanSociety/signet-web/assets/47801291/ca700d29-31ea-4b87-a148-9b1adc10b02b

React-query dev tools cache explorer:
If you are new to react-query, the left side is a list of the client side cache. Each item has an unique query key array as an identifier. For this hook the chain name, isChainSupported and the referendum ID are being used for uniqueness and debugging readability.
On the right side the details of a given query is displayed, including the cached data.
[
![Screenshot 2024-05-21 at 9 46 53](https://github.com/TalismanSociety/signet-web/assets/47801291/df29c374-39a3-487f-8fb0-afc16a2e2346)
](url)
